hapi-fees/src/main/java/com/hedera/services/usage/token/TokenBurnUsage.java

```diff
@@ -36,15 +40,34 @@ public static TokenBurnUsage newEstimate(TransactionBody tokenBurnOp, SigUsage s
 		return new TokenBurnUsage(tokenBurnOp, estimatorFactory.get(sigUsage, tokenBurnOp, ESTIMATOR_UTILS));
 	}
 
+	public TokenBurnUsage givenSubType(SubType subType){
+		this.currentSubType = subType;
+		return this;
+	}
+
 	@Override
 	TokenBurnUsage self() {
 		return this;
 	}
 
 	public FeeData get() {
+		var op = this.op.getTokenBurn();
+
+		if (currentSubType == SubType.TOKEN_NON_FUNGIBLE_UNIQUE) {
+			var bytesToAdd = 0;
+			for (Long o : op.getSerialNumbersList()) {
+				bytesToAdd += o.byteValue();
+			}
+			usageEstimator.addBpt(bytesToAdd);
+			usageEstimator.addRbs(bytesToAdd);
+			var tokenSize = op.getSerialNumbersList().size();
+			usageEstimator.addRbs(tokenEntitySizes.bytesUsedForUniqueTokenTransfers(tokenSize));
+		} else if (currentSubType == SubType.TOKEN_FUNGIBLE_COMMON) {
+			addAmountBpt();
+		}
```

Comment:
Can we introduce a helper function and use it in the rest of the operations as well?

Reply:
I can try, but both queries differ in the type of the repeated argument. I can attempt writing a wrapper for example?